adapters/openx/openx.go

```diff
@@ -140,18 +137,26 @@ func preprocess(imp *openrtb2.Imp, reqExt *openxReqExt) error {
 		imp.BidFloor = openxExt.CustomFloor
 	}
 
+	// outgoing imp.ext should be same as incoming imp.ext minus prebid and bidder
 	impExt := openxImpExt{}
-	addImpExt := false
+	if err := json.Unmarshal(imp.Ext, &impExt); err != nil {
+		return &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+	delete(impExt, "bidder")
+	delete(impExt, "prebid")
 
 	if openxExt.CustomParams != nil {
-		impExt.CustomParams = openxExt.CustomParams
-		addImpExt = true
-	}
-	if bidderExt.AuctionEnvironment != openrtb_ext.ServerSideAuction {
-		impExt.AuctionEnvironment = bidderExt.AuctionEnvironment
-		addImpExt = true
+		var err error
+		if impExt["customParams"], err = json.Marshal(openxExt.CustomParams); err != nil {
+			return &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
```

Comment:
This is correct but we do not have to create error var upfront. Here is how you could do this - 
```
if impExt["customParams"], err := json.Marshal(openxExt.CustomParams); err != nil {
			return &errortypes.BadInput{
				Message: err.Error(),
			}
		}
```

Reply:
The implicit var creation only works with variables; I would get this error:
> non-name impExt["customParams"] on left side of :=